themes/bootstrap3/templates/record/checkbox.phtml

```diff
@@ -1,5 +1,6 @@
+<?php $label = isset($this->context) ? 'select_item_' . $this->context : 'select_item'; ?>
```

Comment:
Development of this file seems incomplete. 

1.) This references a `select_item` key which is undefined in the language files.

2.) I'm not sure where `$this->context` is coming from or if it is ever going to be set. I wonder if a new parameter needs to be added to the Record view helper's getCheckbox method to specify the context.

3.) I also wonder whether we need to have a fallback in case `$label` is set to a value that's not defined in the language files -- this could be done by specifying a default, like `$this->transEscAttr($label, [], $this->translate('select_item'))` which will use the translation for select_item if the specified $label value is not found.

Reply:
Hello @demiankatz ! 
Well, `$this->context` is already available in `function getCheckbox()`:
https://github.com/vufind-org/vufind/blob/f29c765ae1b982cb614ab4f413a225415f899732/module/VuFind/src/VuFind/View/Helper/Root/Record.php#L427
Depending on whether you are in the _result list_ or the _favorites list_ either the token `select_item_search_result` or `select_item_favorite` is generated. But you are right, both tokens are missing in the translation files. I will add them for de and en:
```
select_item_search_result = "Select item for mailing, exporting, printing, saving as favorite or adding to book bag"
select_item_favorite = "Select item for mailing, removing from favorites, exporting, printing or adding to book bag"
```

```
select_item_search_result = "Titel zum Mailen, Exportieren, Drucken, Speichern oder Merken auswählen"
select_item_favorite = "Titel zum Mailen, Löschen, Exportieren, Drucken oder Merken auswählen"
```

Reply:
@elsenhans, the $context variable in the helper method that you linked to is an array of values to add to the template's context, but it does not itself make a value named context accessible to the template. Based on the existing code, it sets $this->id, $this->number, $this->prefix and possibly $this->formAttr, but the array does not contain a 'context' key, so there is no guarantee that $this->context will be set when this method is called, unless it already exists in the view object prior to the call. I'm also not sure where the search_result or favorite values you reference would be coming from. Doing a quick search of the template code, I don't see a context variable being set in a situation where the checkbox template will be included; it only seems to be used for a few narrow things like QR code display. I also suspect that using 'context' as a variable name may not be a good idea (even though there's already a precedent for doing it), because we have a view helper named context -- this may cause some confusion or ambiguity.

Of course, it's possible I'm still missing something somewhere, but this still seems incomplete to me. And if we continue down this road, we also still need to add the generic select_item translation and make sure it gets used as a default if no translation exists for a more specific context-based token, or else we may end up rendering something non-human-readable in certain situations.

Reply:
OK, just talked to @RLangeUni , he still has to do the PR for the helper method. But it should be done in a separate PR.
So, either we wait with this PR for that one or we remove the label in the checkbox here and open another one (or should the label part of your PR @RLangeUni ?).

Reply:
I removed the label and the translations and created a ticket as reminder.
For now we also could use for the aria-label just one token with the translation _"Select item for mailing, exporting, printing, saving as favorite or adding to book bag"_
Just the delete in the list of favorites would be missing, but the most important information would be already available.

Reply:
@elsenhans, I like the idea of a single generic label for now, but I think we should use much less specific text. Every option on the list is configurable and may not be available in every VuFind instance, so listing every possible action is going to be confusing and misleading in many situations. I think it would be better to use a general text like "Select item for further action" and then people can customize this locally to match their actual situation if they wish to. Additionally, if we make the text fully generic, then it will also be usable as a default fallback for other types of checkboxes and will never be wrong. I understand that there are advantages to being very specific, but in the case of the core project, I just don't think there's a reasonable way to do it without a lot of complex logic, and that seems excessive. I'd rather be vague than explicitly wrong.

Reply:
Thanks @demiankatz ! That's a good point with the individual configuration. 
I add your concerns to ticket for the following PR, so they are not lost and the best possible solution can be worked out.